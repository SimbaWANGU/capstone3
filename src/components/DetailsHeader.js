import React from 'react';
import PropTypes from 'prop-types';
import '../styles/DetailsHeader.scss';

function DetailsHeader({ name, photo, team }) {
  return (
    <div className="detailsHeader" data-testid="detailsHeaderDisplay">
      <img src={photo} alt="Scorer" />
      <h1>{name}</h1>
      <p>{team}</p>
    </div>
  );
}

DetailsHeader.propTypes = {
  name: PropTypes.string.isRequired,
  photo: PropTypes.string.isRequired,
  team: PropTypes.string.isRequired,
};

export default DetailsHeader;
