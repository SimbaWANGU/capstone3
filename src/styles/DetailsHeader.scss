$fontColor : rgba(255, 255, 255, 1);

.detailsHeader {
  display: flex;
  flex-direction: column;

  img {
    padding: 10px;
    margin: 10px auto auto auto;
    height: 15vh;
    width: 30vw;
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
    border-radius: 50%;
    object-fit: cover;
  }

  h1 {
    margin: 10px auto auto auto;
    color: $fontColor;
    font-weight: 500;
  }

  p {
    margin: 10px auto auto auto;
    color: $fontColor;
    font-weight: 300;
    font-size: 24px;
  }
}
