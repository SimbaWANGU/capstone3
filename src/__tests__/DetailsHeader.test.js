import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import DetailsHeader from '../components/DetailsHeader';

describe('Details page header section tests', () => {
  const testData = {
    photo: 'https://photo.example.com',
    name: 'Example',
    team: 'Example FC',
  };

  it('Component renders as expected', () => {
    const component = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <DetailsHeader photo={testData.photo} name={testData.name} team={testData.team} />
        </BrowserRouter>
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Component is visible to the user', () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <BrowserRouter>
          <DetailsHeader photo={testData.photo} name={testData.name} team={testData.team} />
        </BrowserRouter>
      </Provider>,
    );
    expect(getByTestId('detailsHeaderDisplay')).toBeInTheDocument();
  });
});
