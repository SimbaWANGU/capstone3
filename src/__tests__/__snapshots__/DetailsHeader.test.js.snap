// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details page header section tests Component renders as expected 1`] = `
<div
  className="detailsHeader"
  data-testid="detailsHeaderDisplay"
>
  <img
    alt="Scorer"
    src="https://photo.example.com"
  />
  <h1>
    Example
  </h1>
  <p>
    Example FC
  </p>
</div>
`;
